src/components/JsonSchemaForm/index.tsx

```diff
@@ -0,0 +1,231 @@
+import { PositiveIntPicker } from "@/components/FormElements/PositiveIntPicker";
+import { LabelHeading } from "@/components/LabelHeading";
+
+export const JsonSchemaForm = ({
+  schema,
+  name,
+}: {
+  schema: JSONSchema7;
+  name: string;
+}) => {
+  const [dereferencedSchema, setDereferencedSchema] =
+    useState<DereferencedSchema>({
+      properties: {},
+      required: [],
+      additionalProperties: false,
+    });
+
+  const [schemaValues, setSchemaValues] = useState<Record<string, string>>({});
+
+  useEffect(() => {
+    const dereferencedSchema = getDereferenceSchema(schema, name);
+    setDereferencedSchema(dereferencedSchema);
+  }, [schema, name]);
+
+  const handleChange = (key: string, value: string) => {
+    setSchemaValues({ ...schemaValues, [key]: value });
+  };
+
+  // key: argument label
+  // prop: argument object
+  //   // value: argument value that includes the key
+  //   // type: argument type
+  //   // description: argument description
+  //   // pattern
+  //   // min-length; max-length
+  const renderComponent = (key: string, prop: any, index?: number) => {
+    const getSpecType = (prop: any) => {
+      if (prop.specType) {
+        return prop.specType;
+      }
+      if (prop.$ref) {
+        return prop.$ref.replace("#/definitions/", "");
+      }
+      return prop.type;
+    };
+
+    const specType = getSpecType(prop);
+    const label = index !== undefined ? `${key}-${index + 1}` : key;
+
+    if (getSpecType(prop)) {
+      switch (specType) {
+        case "Address":
+          return (
+            <Input
+              id={key}
+              key={label}
+              fieldSize="md"
+              label={label}
+              value={schemaValues[label] || ""}
+              error={""}
+              onChange={(e) => {
+                handleChange(label, e.target.value);
+              }}
+              infoText={prop.description || ""}
+              leftElement={<Icon.User03 />}
+            />
+          );
+        case "ScString":
+        case "ScSymbol":
+        case "DataUrl":
+          return (
+            <Input
+              id={key}
+              key={label}
+              fieldSize="md"
+              label={label}
+              value={schemaValues[label] || ""}
+              error={""}
+              onChange={(e) => {
+                handleChange(label, e.target.value);
+              }}
+            />
+          );
+        case "U32":
+        case "U64":
+        case "U128":
+        case "U256":
+          return (
+            <PositiveIntPicker
+              id={key}
+              key={label}
+              label={label}
+              value={schemaValues[label] || ""}
+              error={""}
+              onChange={(e) => {
+                handleChange(label, e.target.value);
+              }}
+            />
+          );
+
+        case "I32":
+        case "I64":
+        case "I128":
+        case "I256":
+          return null;
```

Comment:
Curious: why are we not rendering these types?

Reply:
ah I was going to create a component that allows both positive and negative number. I'll do that next!